Formula/bareos-fd.rb

```diff
@@ -0,0 +1,71 @@
+class BareosFd < Formula
+  desc "Open-Source Data Protection Solution"
+  homepage "https://bareos.org"
+  url "https://github.com/bareos/bareos/archive/Release/17.2.4.tar.gz"
+  sha256 "4c443539012cf5ebb0fdb18878e604e82b951e6429c618acd18762f3c5724799"
+
+  depends_on "readline"
+  depends_on "openssl"
+
+  def install
+    system "./configure",
+    "--prefix=#{prefix}",
+    "--sbindir=#{bin}",
+    "--with-working-dir=#{var}/lib/bareos",
+    "--with-archivedir=#{var}/bareos",
+    "--with-confdir=#{etc}/bareos",
+    "--with-configtemplatedir=#{lib}/bareos/defaultconfigs",
+    "--with-scriptdir=#{lib}/bareos/scripts",
+    "--with-plugindir=#{lib}/bareos/plugins",
+    "--with-fd-password=XXX_REPLACE_WITH_CLIENT_PASSWORD_XXX",
+    "--with-mon-fd-password=XXX_REPLACE_WITH_CLIENT_MONITOR_PASSWORD_XXX",
+    "--with-basename=XXX_REPLACE_WITH_LOCAL_HOSTNAME_XXX",
+    "--with-hostname=XXX_REPLACE_WITH_LOCAL_HOSTNAME_XXX",
+    "--with-python",
+    "--enable-client-only"
+
+    system "make"
+    # * Makefile is intended to create a .pkg file.
+    #   As it is not required or used it's removed.
+    rm "platforms/osx/Makefile"
+    system "make", "install"
+  end
+
+  def post_install
```

Comment:
Are we gaining anything by shelling out here? Could we not handle this on the Homebrew side natively by doing something like:

```ruby
return if File.exist?(etc/"bareos/bareos-fd")

cp Dir[lib/"bareos/defaultconfigs/*"], etc/"bareos"
```

Or:
```ruby
return if File.exist?(etc/"bareos/bareos-fd")

(etc/"bareos").install Dir[lib/"bareos/defaultconfigs/*"]
```
Not sure how ILZ feels. Just seems weird to shell out for something we can handle ourselves with less mess.

Reply:
I'll change it as it is easier to understand what these lines do and delivers exactly the same result 👍

Reply:
Sorry. This will not work. Using copy does not have the same effect as the deploy_config function. deploy_config will also initialize random password and sets the hostname.
I'm not sure if the daemon will start at all, as the template string for the hostname might not be a valid hostname.
So the initial test might fail. If its starts, others can gain access to your system, as it would use a default password.

The deploy_config function is used also on Debian and on Mac, when delivered as PKG file, so I propose to also use it here.

What did indeed look ugly is the `|| true` at the end of the command. deploy_config returns False, if it have not updated the configuration. In this case the system command raises an exception. There might be better ways to cope with this.

Reply:
>If its starts, others can gain access to your system, as it would use a default password.

Even if we do use the `deploy_config`, it seems pretty unacceptable for this package to compromise security if `post_install` happens to fail.

Reply:
What change do you suggest here? If post_install fails, should the daemon not start at all ? This would prevent the daemon from running with default password.

Reply:
@joergsteffens yeah it seems like if `post_install` fails, the daemon shouldn't be able to start at all